core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
```

Comment:
Then I completely do not understand what @jsanda said in his comment. This is turning really complicated really fast. My goal was to reduce the number of metrics idx inserts when possible. And the implementation so far in this PR is just that; have a cache that keeps track of recent metric index inserts and avoid inserting to the metrics index if there is a cache hit.